### PR TITLE
Add dayshift to tsv filenames for reingestion workflows

### LINKED ISSUE #768: Load_data steps for `image` skipped during Wikimedia reingestion
## Description

During a run of `wikimedia_reingestion_workflow`, one of the `pull_data` steps timed out. In the logs, it is reported that **22,080** records were ingested before the task timed out. However, the `load_image_data` steps _were skipped_. This means that partial image data was not reingested for this day.

We should investigate further to understand why partial ingestion is not working in this case.

## Reproduction
In production, look at the `ingest_data_day_shift_1` TaskGroup for the 10-03-22 DagRun of the `wikimedia_reingestion_workflow`.

I have not yet done this, but you can attempt to reproduce this locally by manually raising an error in the `WikimediaCommonsDataIngester` after a large number of records are processed, and then observing the behavior. We should attempt to reproduce this in both the `wikimedia_commons_workflow` and `wikimedia_reingestion_workflow`, to see if it is solely an issue with reingestion.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by stacimc**
## Fixes

Fixes #768 by @stacimc

## Description

This adds the `day_shift` as a suffix to the tsv filenames for reingestion workflows, in order to prevent a race condition where multiple reingestion days running at the same time will attempt to write to a file with the same name. @AetherUnbound did an excellent investigation and write-up of the bug in [this comment](https://github.com/WordPress/openverse-catalog/issues/768#issuecomment-1371463342).

This PR:
* adds the day_shift as a suffix, to ensure sufficient uniqueness. In non-reingestion DAGs, no additional suffix is added.
* removes the `generate_tsv_filenames` task while we're at it. This was a separate task that runs before the `pull_data` step and outputs the tsv filenames to XComs, so they can be grabbed later by the `copy_to_s3` task. For scripts using the `ProviderDataIngester` base class, it does this by initializing the ingester class to grab the file names. The `pull_data` task would then _re_-initialize the ingester class and override its generated filenames with the previously generated ones. Since all the provider scripts are using the ingester class now this should not be necessary.

## Testing Instructions

General things to look for when testing a DAG:
* Verify that the DAG completes successfully. Set the `INGESTION_LIMIT` Airflow variable to a very small number (<= 100) to speed up ingestion.
* Check the logs for the `pull_data` task and verify that you see logs reporting the tsv file paths for each media type the DAG supports.
* Check the logs for the `copy_to_s3` task and verify that it is accessing the same file path reported in the `pull_data` task

Some DAGs to test:
- A regular, non-reingestion DAG, for example `cleveland_museum_workflow`. You should see tsv filenames that do NOT have a `day_shift` appended to the end. Example `pull_data` log:
```
{media.py:183} INFO - Output path: /var/workflow_output/clevelandmuseum_image_v001_20230123230706.tsv
```

- Test with a reingestion DAG, example `wikimedia_reingestion_workflow`. Check the logs for a few different reingestion days and make sure that the appropriate `day_shift` is appended to the filenames. Example `pull_data` log for the `ingest_data_day_shift_4` TaskGroup:
```
{factory_utils.py:45} INFO - Image store location: /var/workflow_output/wikimedia_image_v001_20230123231717_4.tsv
{factory_utils.py:45} INFO - Audio store location: /var/workflow_output/wikimedia_audio_audio_v001_20230123231717_4.tsv
```

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible
      errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by stacimc** [comment]
Drafting to look at extended test failures.

File: openverse_catalog/dags/common/storage/media.py [hunk 1]
```diff
@@ -167,11 +168,16 @@ def _initialize_output_path(
...
+        version = f"v{version or CURRENT_VERSION[self.media_type]}"
+
+        path_components = [
+            provider,
+            self.media_type,
+            version,
+            datetime.now().strftime("%Y%m%d%H%M%S"),
+            tsv_suffix,
+        ]
+        output_file = ("_").join(filter(None, path_components)) + ".tsv"
```
Comment (AetherUnbound) [inline openverse_catalog/dags/common/storage/media.py:180]: Hehe it's a little face 😁 `("_")` (as a real note though I think the parens can be removed):

```suggestion
        output_file = "_".join(filter(None, path_components)) + ".tsv"
```

Awesome use of `filter`!

**Review by AetherUnbound — COMMENTED** [review]
Oh this is rad, I totally forgot that `generate_tsv_filenames` was something we could get rid of now that we're fully converted to the `ProviderDataIngester` derivatives!

This looks good, but when I run this locally both the Wikimedia commons DAG and the Cleveland one gave me XComs that had a suffix value:

```
[2023-01-31, 01:30:16 UTC] {factory_utils.py:44} INFO - Image store location: /var/workflow_output/wikimedia_image_v001_20230131013016_0.tsv
[2023-01-31, 01:30:16 UTC] {factory_utils.py:44} INFO - Audio store location: /var/workflow_output/wikimedia_audio_audio_v001_20230131013016_0.tsv
```

and

```
[2023-01-31, 01:34:33 UTC] {media.py:183} INFO - Output path: /var/workflow_output/clevelandmuseum_image_v001_20230131013433_None.tsv
[2023-01-31, 01:34:33 UTC] {factory_utils.py:44} INFO - Image store location: /var/workflow_output/clevelandmuseum_image_v001_20230131013433_None.tsv
```

Do you know why that's happening? 😮

File: openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py [hunk 2]
```diff
@@ -160,7 +166,7 @@ def _init_media_stores(self) -> dict[str, MediaStore]:
 
         for media_type, provider in self.providers.items():
             StoreClass = get_media_store_class(media_type)
-            media_stores[media_type] = StoreClass(provider)
+            media_stores[media_type] = StoreClass(provider, tsv_suffix=str(day_shift))
```
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py:169]: Oh, it's this line 😅 This is converting `None` and `0` to `'None'` and `'0'` which don't pass the filter! So they're getting appended to the filename anyway.

File: openverse_catalog/dags/common/storage/media.py (hunk 1 above)
Comment (obulat) [reply openverse_catalog/dags/common/storage/media.py:180]: What does this `filter` do?

**Review by obulat — CHANGES_REQUESTED** [review]
I am getting the same issue as @AetherUnbound, in `load_from_s3`:
`Loading image/cleveland_museum/year=2023/clevelandmuseum_image_v001_20230207060449_None.tsv from S3 Bucket openverse-storage into provider_data_image_cleveland_museum_20230101T000000_0`

Can we also add a test for these cases?

**Comment by stacimc** [comment]
Sorry for the delay getting back to this one. Rebased and fixed the identified issue 👍

**Review by AetherUnbound — APPROVED** [review]
I was able to verify, both with standard ingestion & reingestion, that this appropriately appends the day shift when necessary! I have some other comments/suggestions above, but nothing to block a merge 😄

**Pull request merged**